spec/ruby/core/module/refine_spec.rb

```diff
@@ -573,6 +573,43 @@ def foo; end
       end
     end
 
+    ruby_version_is "" ... "2.7" do
+      it "is not honored by Kernel#public_method" do
+        klass = Class.new
+        refinement = Module.new do
+          refine klass do
+            def foo; end
+          end
+        end
+
+        -> {
+          Module.new do
+            using refinement
+            klass.new.public_method(:foo)
+          end
+        }.should raise_error(NameError, /undefined method `foo'/)
+      end
+    end
+
+    ruby_version_is "2.7" do
+      it "is honored by Kernel#public_method" do
+        klass = Class.new
+        refinement = Module.new do
+          refine klass do
+            def foo; end
+          end
+        end
+
+        result = nil
+        Module.new do
+          using refinement
+          result = klass.new.public_method(:foo).class
+        end
+
+        result.should == Method
```

Comment:
it would be simpler do do the check directly above, like:
```
klass.new.public_method(:foo).should be_kind_of(Method)
```

Reply:
I just follow the rules (to be consistent) that are used in other tests in this file:

https://github.com/oracle/truffleruby/blob/9da7552b00072a6c06a0cb1e1a2cce4c9da0bdaa/spec/ruby/core/module/refine_spec.rb#L646

https://github.com/oracle/truffleruby/blob/9da7552b00072a6c06a0cb1e1a2cce4c9da0bdaa/spec/ruby/core/module/refine_spec.rb#L572

I can refactor these 3 tests if you want

Reply:
I see. It's fairly minor, either is fine (keep as-is or refactor the 3 specs)